### DISCUSSION TITLE
[Simulation question]: Scoring Time

### DISCUSSION THREAD

**Discussion by yotaz28** [topas-nbio/TOPAS-nBio#39] [Simulation-question]
### Simulation question

Hi, 

This is related to a problem I had a long time ago that I'm just tackling again now so unfortunately I cannot find the related files. I essentially want to track the timestamps of a particle along its path, especially the points it crosses cell borders and nucleus borders in a TsCellCulture. I was reccomended to use the `b:Sc/MyScorer/IncludeTOPASTime = "True"` setting but last time I tried that it just wasn't adding anything to the scoring for some reason. 

I hope its not too much of an inconvenience but are there any examples of what I'm describing that I could follow? Or are there other ways to achieve what I'm trying to get?

### TOPAS-nBio category

Scoring

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

### Attestation

- [X] I testify that I have fully read the bug reporting guideline beforehand.

**Comment by jschuemann** [comment]
Hi,
I assume you are scoring some track information here? In that case, I recommend using the n-tuple scoring option, where you can define at which interactions you want to write out a customizable set of variables, including the time, either Geant4 or TOPAS time, which in your setting is likely the same. For info on the custom scorers, see:
https://opentopas.readthedocs.io/en/latest/extension-docs/scoring.html

**Reply by yotaz28** [reply]
Hi, sorry about the late reply.

I do plan to write a custom scorer once I figure out exactly what I need for it, but is it not possible to output the time on a phase space scorer? I've found my files where I've had this issue so I've attached them now, and its simply a simulation of 2 spheres of water in a box of more water that are being irradiated with protons, and I'm scoring a few quantities for both spheres using phase space scorers. Including the quantity: 
`b:Sc/MyScorer/IncludeTOPASTime = "True"`

You can see that all other quantities included in the scorers show up but there is no column for the time, am I doing something wrong here?
[data.zip](https://github.com/user-attachments/files/17291326/data.zip)

**Reply by JoseRamosMendez** [reply]
To score the particle's time, use TimeOfFlight

     b:Sc/MyScorer/IncludeTimeOfFlight = "True"
